### DISCUSSION TITLE
Implementing Joomla 'views'

### DISCUSSION THREAD

**Discussion by Ruud68** [Kunena/Kunena-Forum#8902] [Ideas]
Hi,
been testing Kunena 6.
What I noticed is that you 'copied' over the list views (e.g. for categories) from Kunena 5.

I think it would be better to 'migrate' these views to use the Joomla Framework: that way Kunena will fit seamlessly in Joomla, benefit from all code maintenance done in Joomla core.

As a POC I have done some work on the categories view, below the result: main changes are implement the Joomla searchttools and get rid of the theme.min.css (so css is from the joomla template instead of overwritten by custom css):

Is this something you would be interested in?
![image](https://user-images.githubusercontent.com/2733197/162920802-d8c66065-9387-43c0-a4be-212682a917ae.png)

**Comment by xillibit** [comment]
Hello, yes we are insterested on it. If you want work on it you can

**Reply by Ruud68** [reply]
Cool.
If we want to do this right, we must 'part' of the KunenaModel and implement the Joomla models: eg CategoriesModel extends ListModel (instead of KunenaModel). I am now rewritting a KunenaModelBehaviorTrait, that we can then use in (e.g.) CategoriesModel. That way we do not loose Kunena specific methods as we probably need these (not all of them).

So this is a 'bigger' refactor if we want to do this right.

Will do a POC with the Categories view (backend) so we can test and form an opinion etc.

**Comment by Ruud68** [comment]
Okay, just spend a considerable amount of time to get a grip of what needs to change and I have come to the conclusion that this is not going to work in the time I can spent on this.
I think this should be something for kunena version 7 where the Joomla MVC is fully adopted and implemented (basically start from scratch). All model, controller and table classes are currently Kunena own classes with code in it that is obsolete (or not) because that functionality is currently also available in the joomla classes. The long history of Kunena is showing here as well as the HMVC approach it adopted in previous versions.
As an example: when looking at e.g. categories, that means a rewrite of the model and the controller. That links into the category and this needs rewriting of the model / controller and table. Also category is not using (xml) forms but constructing the form in the template file (and thus not benefiting of security / validation / etc.)

How do you see this?

**Reply by 810** [reply]
please create a pr for what you have now. We going to start that as a point for K7.0

**Reply by Ruud68** [reply]
I will do a proper PR for category/categories. Maybe that is something that better fits into J5 which is also planned: So K7 can build on top of that avoiding to have yet again a lot of changes / improvements.

In the meantime I will try to help as much as I can on bugs: for me it is important that K6 goes live :)
